snuba/consumers/consumer_builder.py

```diff
@@ -62,6 +65,33 @@ class MockParameters:
     std_deviation: int
 
 
+def get_sliced_snuba_topic(
+    topic: SnubaTopic, slice_id: Optional[int] = None
+) -> SnubaTopic:
+
+    if slice_id is not None:
+        # there should be some sort of assertion to check
+        # whether the slice_id passed is valid for this storage
+        physical_topic = SLICED_KAFKA_TOPIC_MAP[(topic.value, slice_id)]
+        # this will require registration of the physical topic
+        topic = register_topic(physical_topic)
```

Comment:
let's not reassign `topic`, we could just `return register_topic(physical_topic)` here since we're not doing anything else with it